catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/SourcePollerRunner.java

```diff
@@ -13,247 +13,53 @@
-              .thenComparing(
-                  SourceKey::getOrganization, Comparator.nullsLast(Comparator.naturalOrder())));
-
-  private ExecutorService pool;
-
-  private Map<CachedSource, Lock> sourceStatusThreadLocks = new ConcurrentHashMap<>();
-
-  /**
-   * Creates an empty list of {@link Source} sources to be polled for availability. This constructor
-   * is invoked by the CatalogFramework's blueprint.
-   */
-  public SourcePollerRunner() {
-
-    LOGGER.debug("Creating source poller runner.");
-    sources = new CopyOnWriteArrayList<>();
-  }
-
-  /** Checks the availability of each source in the list of sources to be polled. */
-  @Override
-  public void run() {
-
-    LOGGER.trace("RUNNER checking source statuses");
-
-    for (Source source : sources) {
-
-      if (source != null) {
-
-        checkStatus(source);
-      }
-    }
-  }
-
-  /**
-   * Checks if the specified source is available, updating the internally maintained map of sources
-   * and their status. Lock ensures only one status thread is running per source.
-   *
-   * @param source the source to check if it is available
-   */
-  private void checkStatus(final Source source) {
-
-    if (pool == null) {
-      pool =
-          Executors.newCachedThreadPool(
-              StandardThreadFactoryBuilder.newThreadFactory("sourcePollerRunnerThread"));
-    }
-    final Runnable statusRunner =
-        () -> {
-          final CachedSource cachedSource = cachedSources.get(getSourceKey(source));
-          if (cachedSource != null) {
-            Lock sourceStatusThreadLock = sourceStatusThreadLocks.get(cachedSource);
+import static org.apache.commons.lang3.Validate.notNull;
 
-            if (sourceStatusThreadLock.tryLock()) {
-              try {
-                LOGGER.debug(
-                    "Acquired lock for Source [{}] with id [{}]",
-                    source,
-                    getSourceKey(source).getId());
-                cachedSource.checkStatus();
-              } finally {
-                // release the lock acquired initially
-                sourceStatusThreadLock.unlock();
-                LOGGER.debug(
-                    "Released lock for Source [{}] with id [{}]",
-                    source,
-                    getSourceKey(source).getId());
-              }
-            } else {
-              LOGGER.debug(
-                  "Unable to get lock for Source [{}] with id [{}]."
-                      + "  A status thread is already running.",
-                  source,
-                  getSourceKey(source).getId());
-            }
-          }
-        };
-    pool.execute(statusRunner);
+import com.google.common.collect.ImmutableMap;
+import ddf.catalog.source.Source;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.Validate;
+
+abstract class SourcePollerRunner<V> extends PollerRunner<SourceKey, V> {
```

Comment:
❓What was the reason behind extracting this sub-class from `Poller` since, as far as I know, we don't expect any other `Poller` sub-classes?

Reply:
✏️If we keep this sub-class, we should add JavaDoc comments.

Reply:
I agree that it's not necessary. The motivation was to keep the Source-related logic separate from the Runner-related logic. This is more apparent in the unit tests for these two classes.